ui/lib/sync/addon/components/secrets/page/destinations/destination/sync.ts

```diff
@@ -45,6 +45,11 @@ export default class DestinationSyncPageComponent extends Component<Args> {
   powerSelectAPI: PowerSelectAPI | undefined;
   _lastFetch: KvSecretMetadataModel[] | undefined; // cache the response for filtering purposes
 
+  willDestroy(): void {
+    this.store.clearDataset('sync/association');
+    super.willDestroy();
+  }
+
```

Comment:
We typically do this after the request resolves in the submit method but in this case the user remains on the page and can continue to sync secrets so it made sense to me to fire it when the component is destroyed.